src/main/python/fuct/serialmonitor.py

```diff
@@ -223,6 +223,23 @@ def rip_pages(self, start, end, filepath):
             self.__set_page(page)
             page_data = binascii.hexlify(self.__read_page()).upper()
 
+            # Trim empty flash from start and end of pages
+            first_index = 0
+            for char in page_data:
+                if char != 'F':
```

Comment:
Shouldn't you be checking for FF not just F? Or do I completely fail to read this code? :-)

Reply:
Yes, it really should be looking for FF. That implementation is looping over each nibble of the ASCII character string, finding the first non 'F' character then keeping the index of the character before that. The first index is "rounded" to 16 byte alignment so that is OK but the last index is not being kept to byte alignment which is a screw up.
I think it would be better to loop with a bytearray instead. No byte alignment problems and can search for '\xff', which is clearer.